src/Uno.UI/UI/Xaml/Internal/DependencyPropertyHelper.cs

```diff
@@ -0,0 +1,216 @@
+#nullable enable
+
+using Microsoft.UI.Xaml;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Uno.UI.Xaml.Core;
+
+/// <summary>
+/// The goal of this class is to provide a set of helper methods to work with <see cref="DependencyProperty"/> from
+/// external projects.
+/// </summary>
+/// <remarks>
+/// A small reflection is still required to access the class because those using it needs to know what they are doing.
+/// </remarks>
+internal static class DependencyPropertyHelper
+{
+	/// <summary>
+	/// Get the default value of a <see cref="DependencyProperty"/> for a given type.
+	/// </summary>
+	/// <remarks>
+	/// This is the value defined in the metadata of the property, which _may_ be overriden
+	/// using the internal DependencyProperty.OverrideMetadata() method on a per-type basis
+	/// -- that's why the type parameter is required.
+	///
+	/// A classic example of that is the Button.VerticalAlignmentProperty which is overridden
+	/// to be VerticalAlignment.Center, while the default value is VerticalAlignment.Stretch
+	/// from the base class.
+	/// </remarks>
+	public static object? GetDefaultValue(DependencyProperty dependencyProperty, Type forType)
+		=> dependencyProperty.GetMetadata(forType)?.DefaultValue;
+
+	/// <summary>
+	/// Get the default value of a <see cref="DependencyProperty"/> for a given type.
+	/// </summary>
+	/// <remarks>
+	/// This is the value defined in the metadata of the property, which _may_ be overriden
+	/// using the internal DependencyProperty.OverrideMetadata() method on a per-type basis
+	/// -- that's why the type parameter is required.
+	///
+	/// A classic example of that is the Button.VerticalAlignmentProperty which is overridden
+	/// to be VerticalAlignment.Center, while the default value is VerticalAlignment.Stretch
+	/// from the base class.
+	/// </remarks>
+	public static object? GetDefaultValue<T>(DependencyProperty dependencyProperty)
+		where T : DependencyObject
+		=> dependencyProperty.GetMetadata(typeof(T))?.DefaultValue;
```

Comment:
Same here, both `GetMetadata` and `DefaultValue` are available publicly. It will be easier anyways to do the calls directly rather than using reflection or unsafe accessor